ts/segment.go

```diff
@@ -40,12 +40,12 @@ type Segment struct {
 }
 
 // Len returns the length of the head and tail.
-func (s *Segment) Len() int {
+func (s Segment) Len() int {
 	return len(s.Head) + len(s.Tail)
 }
 
 // Finalize will release resources kept by the segment if pooled.
-func (s *Segment) Finalize() {
```

Comment:
Yeah. But we always use it as a value type everywhere, so to actually call `Finalize()` one would have to write something like `s := stream.Segment(); (&s).Finalize()` otherwise. Thoughts?